.github/workflows/node.js.yml

```diff
@@ -28,3 +28,17 @@ jobs:
     - run: npm test
       env:
         FORCE_COLOR: 3
+
+  # This succeeds if all the matrix builds succeed. This allows us to set branch
+  # protection rules on this job in particular, rather than having to change
+  # branch protection rules when the matrix changes. See
+  # https://github.community/t/status-check-for-a-matrix-jobs/127354/7.
+  testall:
+    if: ${{ always() }}
+    runs-on: ubuntu-latest
+    name: Test (matrix)
+    needs: test
```

Comment:
Given this job depenency I don't think the step checking for success should be necessary. If the matrix job fails, this job should just never run. Not sure if specifying no steps at all is a problem though?

Reply:
Based on your other comments, I'm going to eliminate the need for this job, but, for the record:

The idea here is to ensure that 1) the summary runs after the completion of the matrix builds and 2) it always runs, whether the matrix builds passed or failed. This means we get a definitive pass or fail.

Ordinarily, if a dependency fails, then GitHub Actions skips the dependent job, putting it into that gray indeterminate state. But the gray state doesn't count as a failure and won't block a merge if selected as part of the branch protection rules.